### DISCUSSION TITLE
Is it possible to recover illumination.

### DISCUSSION THREAD

**Discussion by EMwavelength** [mitsuba-renderer/mitsuba3#217] [Q&A] [answered]
Hi,
   Is it possible to recover unknown illumination observed in an image,i.e a diffuse object lit by sunlight? I know it's a big problem in graphics to recover unknown illumination, for inverse rendering illumination must be known, is mitsuba able to solve such problem?

**Comment by Speierers** [comment, marked as answer]
Yes it is be possible. 

Mitsuba 3 is able to produce gradients for the parameters of emitters (e.g. area light, point light, envmap, ...). It is therefore possible to write a simple optimization pipeline that optimizes for the texture data of an environment light for instance.

**Reply by EMwavelength** [reply]
Hi,
   I tried your suggestion, but it seems the optimizer is unable to recover direct sunlight, don't know whats causing it, please guide me in right direction.

Target:
![target](https://user-images.githubusercontent.com/111644839/187195737-2965c46e-377a-4b8e-a31c-ee71bc16a0f1.jpg)

Optimized:
![optimized](https://user-images.githubusercontent.com/111644839/187195805-a5faef99-2135-4987-a659-91f3bafe9d2a.jpg)

TargetEnvmap:
![targetenvmap](https://user-images.githubusercontent.com/111644839/187195848-318bf6c9-707d-4702-a65a-3aa8194343a6.jpg)

OptimizedEnvmap:
![optimizedenvmap](https://user-images.githubusercontent.com/111644839/187195892-e0c4dfb0-ed96-49ce-bf59-21d88691aa3e.jpg)

**Reply by EMwavelength** [reply]
Hi,
   Thank you for the answer your suggestions works to some good degree after i run optimization a little bit more.
Optimizationcorrected:
![optimizedcorrected](https://user-images.githubusercontent.com/111644839/187214488-6137e14b-bc66-48d6-b614-7f3038171cf1.jpg)
TargetOverExposed:
![targetoverexposed](https://user-images.githubusercontent.com/111644839/187214726-2fb3888a-939e-4d91-8ce0-daca1c2e0944.jpg)
